From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MrHua269 <mrhua269@gmail.com>
Date: Sat, 18 Apr 2026 09:03:56 +0800
Subject: [PATCH] Luminol: Configurable region format framework

This patch contains implementations that are based on the following projects:

Region format framework & Buffered Linear
Original license: GPL-3.0-only
Original project: https://github.com/LuminolMC/Luminol

Updated against ref-sources/0002-Configurable-region-format-framework.patch
(newer upstream 26.2 implementation), retaining the me.earthme.luminol package.
Supports MCA and Buffered Linear. The independent Linear v2 format is removed.
The 26.3 Optional region cache and read/create semantics are preserved.
The Buffered Linear flusher closes after Moonrise region I/O has drained.

diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/io/ChunkSystemRegionFileStorage.java b/ca/spottedleaf/moonrise/patches/chunk_system/io/ChunkSystemRegionFileStorage.java
index a814512fcfb85312474ae2c2c21443843bf57831..26ce425c444b19ed6f4be117c57f8fe20db7d0cc 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/io/ChunkSystemRegionFileStorage.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/io/ChunkSystemRegionFileStorage.java
@@ -8,9 +8,9 @@ public interface ChunkSystemRegionFileStorage {
 
     public boolean moonrise$doesRegionFileNotExistNoIO(final int chunkX, final int chunkZ);
 
-    public RegionFile moonrise$getRegionFileIfLoaded(final int chunkX, final int chunkZ);
+    public me.earthme.luminol.data.RegionFile moonrise$getRegionFileIfLoaded(final int chunkX, final int chunkZ); // Luminol - Configurable region file format
 
-    public RegionFile moonrise$getRegionFileIfExists(final int chunkX, final int chunkZ) throws IOException;
+    public me.earthme.luminol.data.RegionFile moonrise$getRegionFileIfExists(final int chunkX, final int chunkZ) throws IOException; // Luminol - Configurable region file format
 
     public MoonriseRegionFileIO.RegionDataController.WriteData moonrise$startWrite(
             final int chunkX, final int chunkZ, final CompoundTag compound
diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/io/MoonriseRegionFileIO.java b/ca/spottedleaf/moonrise/patches/chunk_system/io/MoonriseRegionFileIO.java
index 3bb9b58ee97464687e348e23b99159226415c267..d311f5004cdb5023a9ea1239a2018152aa2f9a13 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/io/MoonriseRegionFileIO.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/io/MoonriseRegionFileIO.java
@@ -1273,7 +1273,7 @@ public final class MoonriseRegionFileIO {
                 this.regionDataController.finishWrite(this.chunkX, this.chunkZ, writeData);
                 // Paper start - flush regionfiles on save
                 if (this.world.paperConfig().chunks.flushRegionsOnSave) {
-                    final RegionFile regionFile = this.regionDataController.getCache().moonrise$getRegionFileIfLoaded(this.chunkX, this.chunkZ);
+                    final me.earthme.luminol.data.RegionFile regionFile = this.regionDataController.getCache().moonrise$getRegionFileIfLoaded(this.chunkX, this.chunkZ); // Luminol - Add configurable region file
                     if (regionFile != null) {
                         regionFile.flush();
                     } // else: evicted from cache, which should have called flush
@@ -1489,7 +1489,7 @@ public final class MoonriseRegionFileIO {
 
         public static interface IORunnable {
 
-            public void run(final RegionFile regionFile) throws IOException;
+            public void run(final me.earthme.luminol.data.RegionFile regionFile) throws IOException; // Luminol - Configurable region file format
 
         }
     }
diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/storage/ChunkSystemChunkBuffer.java b/ca/spottedleaf/moonrise/patches/chunk_system/storage/ChunkSystemChunkBuffer.java
index 51c126735ace8fdde89ad97b5cab62f244212db0..f30cf8153bf9ac51c5676518f0c68f3efefc7ece 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/storage/ChunkSystemChunkBuffer.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/storage/ChunkSystemChunkBuffer.java
@@ -8,5 +8,5 @@ public interface ChunkSystemChunkBuffer {
 
     public void moonrise$setWriteOnClose(final boolean value);
 
-    public void moonrise$write(final RegionFile regionFile) throws IOException;
+    public void moonrise$write(final me.earthme.luminol.data.RegionFile regionFile) throws IOException; // Luminol - Configurable region file format
 }
diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index de0194433118619b2eb0b76007cbe605de75b5a0..8150c89397ffa82bfa625634936d6b94a98b2cec 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1122,6 +1122,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         LOGGER.info("Waiting for all RegionFile I/O tasks to complete...");
         ca.spottedleaf.moonrise.patches.chunk_system.io.MoonriseRegionFileIO.flush((MinecraftServer)(Object)this);
         LOGGER.info("All RegionFile I/O tasks to complete");
+        org.dreeam.leaf.config.modules.misc.RegionFormatConfig.closeFlusherIfEnabled(); // Luminol - Configurable region file format
         if ((Object)this instanceof net.minecraft.server.dedicated.DedicatedServer) {
             ca.spottedleaf.moonrise.common.util.MoonriseCommon.haltExecutors();
         }
diff --git a/net/minecraft/util/worldupdate/FileToUpgrade.java b/net/minecraft/util/worldupdate/FileToUpgrade.java
index a7f2cfa9277c898038f6b9e0a3401db51012b64c..c794fc9af3fba2cf4df53641f47623a781ccc06f 100644
--- a/net/minecraft/util/worldupdate/FileToUpgrade.java
+++ b/net/minecraft/util/worldupdate/FileToUpgrade.java
@@ -4,5 +4,5 @@ import java.util.List;
 import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.chunk.storage.RegionFile;
 
-public record FileToUpgrade(RegionFile file, List<ChunkPos> chunksToUpgrade) {
+public record FileToUpgrade(me.earthme.luminol.data.RegionFile file, List<ChunkPos> chunksToUpgrade) { // Luminol - Configurable region file format
 }
diff --git a/net/minecraft/util/worldupdate/RegionStorageUpgrader.java b/net/minecraft/util/worldupdate/RegionStorageUpgrader.java
index b228cd6c54e8c8998923ab332882d99092769c67..e8f252cd67fdc13ec4b57278878541a09447c998 100644
--- a/net/minecraft/util/worldupdate/RegionStorageUpgrader.java
+++ b/net/minecraft/util/worldupdate/RegionStorageUpgrader.java
@@ -36,7 +36,7 @@ import org.slf4j.Logger;
 public class RegionStorageUpgrader {
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final String NEW_DIRECTORY_PREFIX = "new_";
-    private static final Pattern REGEX = Pattern.compile("^r\\.(-?[0-9]+)\\.(-?[0-9]+)\\.mca$");
+    private static final Pattern REGEX = Pattern.compile("^r\\.(-?[0-9]+)\\.(-?[0-9]+)\\." + org.dreeam.leaf.config.modules.misc.RegionFormatConfig.regionFormat.getExtensionName() + "$"); // Luminol - Configurable region file format
     private final DataFixer dataFixer;
     private final UpgradeProgress upgradeProgress;
     private final String type;
@@ -162,7 +162,7 @@ public class RegionStorageUpgrader {
     }
 
     private static List<FileToUpgrade> getAllChunkPositions(final RegionStorageInfo info, final Path regionFolder) {
-        File[] files = regionFolder.toFile().listFiles((dir, name) -> name.endsWith(".mca"));
+        File[] files = regionFolder.toFile().listFiles((dir, name) -> name.endsWith(net.minecraft.world.level.chunk.storage.RegionFileStorage.getExtensionName())); // Luminol - Configurable region file format
         if (files == null) {
             return List.of();
         }
@@ -176,7 +176,8 @@ public class RegionStorageUpgrader {
                 int zOffset = Integer.parseInt(regex.group(2)) << 5;
                 List<ChunkPos> chunkPositions = Lists.newArrayList();
 
-                try (RegionFile regionSource = new RegionFile(info, regionFile.toPath(), regionFolder, true)) {
+                me.earthme.luminol.utils.RegionCreatorInfo regionFileInfo = new me.earthme.luminol.utils.RegionCreatorInfo(info, regionFile.toPath(), regionFolder, true); // Luminol - Configurable region file format
+                try (me.earthme.luminol.data.RegionFile regionSource = org.dreeam.leaf.config.modules.misc.RegionFormatConfig.regionFormat.getCreator().create(regionFileInfo)) { // Luminol - Configurable region file format
                     for (int x = 0; x < 32; x++) {
                         for (int z = 0; z < 32; z++) {
                             ChunkPos pos = new ChunkPos(x + xOffset, z + zOffset);
@@ -253,7 +254,7 @@ public class RegionStorageUpgrader {
         return storage.upgradeChunkTag(chunkTag, this.defaultVersion, this.dataFixContextTag, targetVersion);
     }
 
-    private void onFileFinished(final RegionFile regionFile) {
+    private void onFileFinished(final me.earthme.luminol.data.RegionFile regionFile) { // Luminol - Configurable region file format
         if (this.recreateRegionFiles) {
             if (this.previousWriteFuture != null) {
                 this.previousWriteFuture.join();
diff --git a/net/minecraft/world/level/chunk/storage/RegionFile.java b/net/minecraft/world/level/chunk/storage/RegionFile.java
index 1616ab8a89602a3cdc00a9b8608e42f316356273..fe7cd2d344f4af48fb179654d74b55d083f58bba 100644
--- a/net/minecraft/world/level/chunk/storage/RegionFile.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFile.java
@@ -22,7 +22,7 @@ import net.minecraft.world.level.ChunkPos;
 import org.jspecify.annotations.Nullable;
 import org.slf4j.Logger;
 
-public class RegionFile implements AutoCloseable, ca.spottedleaf.moonrise.patches.chunk_system.storage.ChunkSystemRegionFile { // Paper - rewrite chunk system
+public class RegionFile implements AutoCloseable, ca.spottedleaf.moonrise.patches.chunk_system.storage.ChunkSystemRegionFile, me.earthme.luminol.data.RegionFile { // Paper - rewrite chunk system // Luminol - Configurable region file format
     private static final Logger LOGGER = LogUtils.getLogger();
     public static final int MAX_CHUNK_SIZE = 500 * 1024 * 1024; // Paper - don't write garbage data to disk if writing serialization fails
     private static final int SECTOR_BYTES = 4096;
@@ -907,7 +907,7 @@ public class RegionFile implements AutoCloseable, ca.spottedleaf.moonrise.patche
         }
 
         @Override
-        public final void moonrise$write(final RegionFile regionFile) throws IOException {
+        public final void moonrise$write(final me.earthme.luminol.data.RegionFile regionFile) throws IOException { // Luminol - Configurable region file format
             regionFile.write(this.pos, ByteBuffer.wrap(this.buf, 0, this.count));
         }
         // Paper end - rewrite chunk system
diff --git a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
index d8d2d100e551b363cab10755a281c46874bdbca3..7d31f3a8941af490c803a9203df3654563413c13 100644
--- a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -22,7 +22,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
     private static final org.slf4j.Logger LOGGER = com.mojang.logging.LogUtils.getLogger(); // Paper
     public static final String ANVIL_EXTENSION = ".mca";
     private static final int MAX_CACHE_SIZE = 256;
-    private final Long2ObjectLinkedOpenHashMap<Optional<RegionFile>> regionCache = new Long2ObjectLinkedOpenHashMap<>();
+    private final Long2ObjectLinkedOpenHashMap<Optional<me.earthme.luminol.data.RegionFile>> regionCache = new Long2ObjectLinkedOpenHashMap<>(); // Luminol - Configurable region file format
     private final RegionStorageInfo info;
     private final Path folder;
     private final boolean sync;
@@ -33,7 +33,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
     @Nullable
     public static ChunkPos getRegionFileCoordinates(Path file) {
         String fileName = file.getFileName().toString();
-        if (!fileName.startsWith("r.") || !fileName.endsWith(".mca")) {
+        if (!fileName.startsWith("r.") || !fileName.endsWith(getExtensionName())) { // Luminol - Configurable region file format
             return null;
         }
 
@@ -58,8 +58,26 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
     private static final int MAX_NON_EXISTING_CACHE = 1024 * 4;
     private final it.unimi.dsi.fastutil.longs.LongLinkedOpenHashSet nonExistingRegionFiles = new it.unimi.dsi.fastutil.longs.LongLinkedOpenHashSet();
     private static String getRegionFileName(final int chunkX, final int chunkZ) {
-        return "r." + (chunkX >> REGION_SHIFT) + "." + (chunkZ >> REGION_SHIFT) + ".mca";
+        return "r." + (chunkX >> REGION_SHIFT) + "." + (chunkZ >> REGION_SHIFT) + getExtensionName(); // Luminol - Configurable region file format
     }
+    // Luminol start - Configurable region file format
+    public static me.earthme.luminol.data.RegionFile createNew(final RegionStorageInfo info, final Path filePath, final Path folder, final boolean sync) throws IOException{
+        final me.earthme.luminol.enums.EnumRegionFormat regionFormat = org.dreeam.leaf.config.modules.misc.RegionFormatConfig.regionFormat;
+        final String fullFileName = filePath.getFileName().toString();
+        final String[] fullNameSplit = fullFileName.split("\\.");
+        final String extensionName = fullNameSplit[fullNameSplit.length - 1];
+
+        if (!regionFormat.getExtensionName().equalsIgnoreCase(extensionName)) {
+            throw new IOException("Invalid region file format: " + extensionName + " expected " + regionFormat.getExtensionName());
+        }
+
+        return regionFormat.getCreator().create(new me.earthme.luminol.utils.RegionCreatorInfo(info, filePath, folder, sync));
+    }
+
+    public static String getExtensionName() {
+        return "." + org.dreeam.leaf.config.modules.misc.RegionFormatConfig.regionFormat.getExtensionName();
+    }
+    // Luminol end - Configurable region file format
 
     private boolean doesRegionFilePossiblyExist(final long position) {
         synchronized (this.nonExistingRegionFiles) {
@@ -93,13 +111,13 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
     }
 
     @Override
-    public synchronized final RegionFile moonrise$getRegionFileIfLoaded(final int chunkX, final int chunkZ) {
-        final Optional<RegionFile> ret = this.regionCache.getAndMoveToFirst(ChunkPos.pack(chunkX >> REGION_SHIFT, chunkZ >> REGION_SHIFT));
+    public synchronized final me.earthme.luminol.data.RegionFile moonrise$getRegionFileIfLoaded(final int chunkX, final int chunkZ) { // Luminol - Configurable region file format
+        final Optional<me.earthme.luminol.data.RegionFile> ret = this.regionCache.getAndMoveToFirst(ChunkPos.pack(chunkX >> REGION_SHIFT, chunkZ >> REGION_SHIFT)); // Luminol - Configurable region file format
         return ret == null || ret.isEmpty() ? null : ret.get();
     }
 
     @Override
-    public synchronized final RegionFile moonrise$getRegionFileIfExists(final int chunkX, final int chunkZ) throws IOException {
+    public synchronized final me.earthme.luminol.data.RegionFile moonrise$getRegionFileIfExists(final int chunkX, final int chunkZ) throws IOException { // Luminol - Configurable region file format
         return this.getRegionFile(new ChunkPos(chunkX, chunkZ), false);
     }
 
@@ -115,7 +133,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         }
 
         final ChunkPos pos = new ChunkPos(chunkX, chunkZ);
-        final RegionFile regionFile = this.getRegionFile(pos, true);
+        final me.earthme.luminol.data.RegionFile regionFile = this.getRegionFile(pos, true); // Luminol - Configurable region file format
 
         // note: not required to keep regionfile loaded after this call, as the write param takes a regionfile as input
         // (and, the regionfile parameter is unused for writing until the write call)
@@ -149,7 +167,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
     ) throws IOException {
         final ChunkPos pos = new ChunkPos(chunkX, chunkZ);
         if (writeData.result() == ca.spottedleaf.moonrise.patches.chunk_system.io.MoonriseRegionFileIO.RegionDataController.WriteData.WriteResult.DELETE) {
-            final RegionFile regionFile = this.moonrise$getRegionFileIfExists(chunkX, chunkZ);
+            final me.earthme.luminol.data.RegionFile regionFile = this.moonrise$getRegionFileIfExists(chunkX, chunkZ); // Luminol - Configurable region file format
             if (regionFile != null) {
                 regionFile.clear(pos);
             } // else: didn't exist
@@ -164,7 +182,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
     public final ca.spottedleaf.moonrise.patches.chunk_system.io.MoonriseRegionFileIO.RegionDataController.ReadData moonrise$readData(
         final int chunkX, final int chunkZ
     ) throws IOException {
-        final RegionFile regionFile = this.moonrise$getRegionFileIfExists(chunkX, chunkZ);
+        final me.earthme.luminol.data.RegionFile regionFile = this.moonrise$getRegionFileIfExists(chunkX, chunkZ); // Luminol - Configurable region file format
 
         final DataInputStream input = regionFile == null ? null : regionFile.getChunkDataInputStream(new ChunkPos(chunkX, chunkZ));
 
@@ -209,7 +227,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
 
             final ChunkPos pos = new ChunkPos(chunkX, chunkZ);
             final ChunkPos headerChunkPos = SerializableChunkData.getChunkCoordinate(ret);
-            final RegionFile regionFile = this.getRegionFile(pos, true);
+            final me.earthme.luminol.data.RegionFile regionFile = this.getRegionFile(pos, true); // Luminol - Configurable region file format
 
             if (regionFile.getRecalculateCount() != readData.recalculateCount()) {
                 return null;
@@ -240,12 +258,12 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         this.isChunkData = info.dfuType()[0] == net.minecraft.util.datafix.DataFixTypes.CHUNK; // Paper - recalculate region file headers
     }
 
-    private @Nullable RegionFile getRegionFile(final ChunkPos pos, final boolean create) throws IOException {
+    private me.earthme.luminol.data.@Nullable RegionFile getRegionFile(final ChunkPos pos, final boolean create) throws IOException { // Luminol - Configurable region file format
         // Paper start - rewrite chunk system
         synchronized (this) {
             final long key = ChunkPos.pack(pos.x() >> REGION_SHIFT, pos.z() >> REGION_SHIFT);
 
-            final Optional<RegionFile> cached = this.regionCache.getAndMoveToFirst(key);
+            final Optional<me.earthme.luminol.data.RegionFile> cached = this.regionCache.getAndMoveToFirst(key); // Luminol - Configurable region file format
             if (cached != null) {
                 if (cached.isPresent()) {
                     return cached.get();
@@ -272,7 +290,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
             final int cacheSize = io.papermc.paper.configuration.GlobalConfiguration.get() == null ? 256 : io.papermc.paper.configuration.GlobalConfiguration.get().misc.regionFileCacheSize; // Paper - Sanitise RegionFileCache and make configurable - Config not available during initial FileFixerUpper run
 
             if (this.regionCache.size() >= cacheSize) {
-                final Optional<RegionFile> evicted = this.regionCache.removeLast();
+                final Optional<me.earthme.luminol.data.RegionFile> evicted = this.regionCache.removeLast(); // Luminol - Configurable region file format
                 if (evicted.isPresent()) {
                     evicted.get().close();
                 }
@@ -282,7 +300,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
 
             FileUtil.createDirectoriesSafe(this.folder);
 
-            final RegionFile ret = new RegionFile(this.info, regionPath, this.folder, this.sync);
+            final me.earthme.luminol.data.RegionFile ret = createNew(this.info, regionPath, this.folder, this.sync); // Luminol - Configurable region file format
 
             this.regionCache.putAndMoveToFirst(key, Optional.of(ret));
 
@@ -291,7 +309,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         // Paper end - rewrite chunk system
     }
 
-    private RegionFile getOrCreateRegionFile(final ChunkPos pos) throws IOException {
+    private me.earthme.luminol.data.RegionFile getOrCreateRegionFile(final ChunkPos pos) throws IOException { // Luminol - Configurable region file format
         return Objects.requireNonNull(this.getRegionFile(pos, true));
     }
 
@@ -300,7 +318,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         LOGGER.error("{} ({} - {},{}) Go clean it up to remove this message. /minecraft:tp {} 128 {} - DO NOT REPORT THIS TO PAPER OR LEAF - You may ask for help on Discord, but do not file an issue. These error messages can not be removed. - If you think this is a Leaf bug, please report it at https://github.com/Winds-Studio/Leaf/issues", msg, file.toString().replaceAll(".+[\\\\/]", ""), x, z, x << 4, z << 4); // Leaf - Rebrand
     }
 
-    private static CompoundTag readOversizedChunk(RegionFile regionfile, ChunkPos chunkCoordinate) throws IOException {
+    private static CompoundTag readOversizedChunk(me.earthme.luminol.data.RegionFile regionfile, ChunkPos chunkCoordinate) throws IOException { // Luminol - Configurable region file format
         synchronized (regionfile) {
             try (DataInputStream datainputstream = regionfile.getChunkDataInputStream(chunkCoordinate)) {
                 CompoundTag oversizedData = regionfile.getOversizedData(chunkCoordinate.x(), chunkCoordinate.z());
@@ -333,7 +351,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
     // Paper end
 
     public @Nullable CompoundTag read(final ChunkPos pos) throws IOException {
-        RegionFile region = this.getRegionFile(pos, false);
+        me.earthme.luminol.data.RegionFile region = this.getRegionFile(pos, false); // Luminol - Configurable region file format
         if (region == null) {
             return null;
         }
@@ -368,7 +386,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
     }
 
     public void scanChunk(final ChunkPos pos, final StreamTagVisitor scanner) throws IOException {
-        RegionFile region = this.getRegionFile(pos, false);
+        me.earthme.luminol.data.RegionFile region = this.getRegionFile(pos, false); // Luminol - Configurable region file format
         if (region != null) {
             try (DataInputStream regionChunkInputStream = region.getChunkDataInputStream(pos)) {
                 if (regionChunkInputStream != null) {
@@ -386,7 +404,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         }
         // Paper end - rewrite chunk system
         if (!SharedConstants.DEBUG_DONT_SAVE_WORLD) {
-            RegionFile region = this.getOrCreateRegionFile(pos);
+            me.earthme.luminol.data.RegionFile region = this.getOrCreateRegionFile(pos); // Luminol - Configurable region file format
             if (value == null) {
                 region.clear(pos);
             } else {
@@ -412,7 +430,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         // Paper start - rewrite chunk system
         synchronized (this) {
             final ExceptionCollector<IOException> exceptionCollector = new ExceptionCollector<>();
-            for (final Optional<RegionFile> entry : this.regionCache.values()) {
+            for (final Optional<me.earthme.luminol.data.RegionFile> entry : this.regionCache.values()) { // Luminol - Configurable region file format
                 if (entry.isEmpty()) {
                     continue;
                 }
@@ -431,7 +449,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         // Paper start - rewrite chunk system
         synchronized (this) {
             final ExceptionCollector<IOException> exceptionCollector = new ExceptionCollector<>();
-            for (final Optional<RegionFile> entry : this.regionCache.values()) {
+            for (final Optional<me.earthme.luminol.data.RegionFile> entry : this.regionCache.values()) { // Luminol - Configurable region file format
                 if (entry.isEmpty()) {
                     continue;
                 }
